velox/experimental/cudf/exec/Utilities.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/experimental/cudf/exec/Utilities.h"
+
+#include <cudf/concatenate.hpp>
+#include <cudf/detail/utilities/stream_pool.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/mr/device/arena_memory_resource.hpp>
+#include <rmm/mr/device/cuda_async_memory_resource.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+#include <rmm/mr/device/managed_memory_resource.hpp>
+#include <rmm/mr/device/owning_wrapper.hpp>
+#include <rmm/mr/device/pool_memory_resource.hpp>
+
+#include <common/base/Exceptions.h>
+
+#include <cstdlib>
+#include <memory>
+#include <string_view>
+
+namespace facebook::velox::cudf_velox {
+
+namespace {
+[[nodiscard]] auto makeCudaMr() {
+  return std::make_shared<rmm::mr::cuda_memory_resource>();
+}
+
+[[nodiscard]] auto makePoolMr() {
+  return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(
+      makeCudaMr(), rmm::percent_of_free_device_memory(50));
+}
+
+[[nodiscard]] auto makeAsyncMr() {
+  return std::make_shared<rmm::mr::cuda_async_memory_resource>();
+}
+
+[[nodiscard]] auto makeManagedMr() {
+  return std::make_shared<rmm::mr::managed_memory_resource>();
+}
+
+[[nodiscard]] auto makeArenaMr() {
+  return rmm::mr::make_owning_wrapper<rmm::mr::arena_memory_resource>(
+      makeCudaMr());
+}
+
+[[nodiscard]] auto makeManagedPoolMr() {
+  return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(
+      makeManagedMr(), rmm::percent_of_free_device_memory(50));
+}
+} // namespace
+
+std::shared_ptr<rmm::mr::device_memory_resource> createMemoryResource(
+    std::string_view mode) {
```

Comment:
const std::string&, we don't need to convert the string to std::string_view

Reply:
String to string_view conversions happen automatically and are generally a nicer way to pass strings to functions.